coffea/jetmet_tools/CorrectedJetsFactory.py

```diff
@@ -114,53 +150,65 @@ def __init__(self, name_map, jec_stack):
         if "massRaw" not in name_map or name_map["massRaw"] is None:
             warnings.warn(
                 "There is no name mapping for massRaw,"
-                " CorrectedJets will assume that <object>.mass is raw pt!"
+                " CorrectedJets will assume that <object>.mass is raw mass!"
+            )
+            name_map["massRaw"] = name_map["JetMass"] + "_raw"
+
+        self.jec_stack = jec_stack
+        self.name_map = name_map
+
+        if self.jec_stack.use_clib:
+            # For clib scenario, load corrections from json_path
+            self.load_corrections_clib()
+        else:
+            # For non-clib scenario, use the provided corrections (e.g., JEC/JER)
+            self.load_corrections_jecstack()
+
+        if "ptGenJet" not in name_map:
+            warnings.warn(
+                'Input JaggedCandidateArray must have "ptGenJet" in order to apply hybrid JER smearing method. Stochastic smearing will be applied.'
             )
-            name_map["ptRaw"] = name_map["JetMass"] + "_raw"
+            self.forceStochastic = True
 
+    def load_corrections_clib(self):
+        """Load the corrections from correctionlib using the json_path in JECStack."""
+        self.corrections = self.jec_stack.corrections
+
+    def load_corrections_jecstack(self):
+        """Use the corrections provided in the JECStack for non-clib scenario."""
+        self.corrections = self.jec_stack.corrections
+
+        # Ensure all required inputs have mappings
         total_signature = set()
         for part in _stack_parts:
-            attr = getattr(jec_stack, part)
+            attr = getattr(self.jec_stack, part)
             if attr is not None:
                 total_signature.update(attr.signature)
 
-        missing = total_signature - set(name_map.keys())
+        missing = total_signature - set(self.name_map.keys())
         if len(missing) > 0:
             raise Exception(
                 f"Missing mapping of {missing} in name_map!"
                 + " Cannot evaluate jet corrections!"
                 + " Please supply mappings for these variables!"
             )
 
-        if "ptGenJet" not in name_map:
-            warnings.warn(
-                'Input JaggedCandidateArray must have "ptGenJet" in order to apply hybrid JER smearing method. Stochastic smearing will be applied.'
-            )
-            self.forceStochastic = True
-
-        self.real_sig = [v for k, v in name_map.items()]
-        self.name_map = name_map
-        self.jec_stack = jec_stack
-
-    def uncertainties(self):
```

Comment:
why was this function removed?

Justify and alter tests or please add it back!